### PR TITLE
langchain[patch]: Fix run on dataset method for built-in evaluators

### PR DESCRIPTION
**Pull request by jacoblee93**
CC @agola11 @hinthornw

### CODE REVIEW

File: langchain/src/smith/runner_utils.ts [hunk 1]
```diff
@@ -59,6 +59,30 @@ class RunIdExtractor {
...
+
+  constructor() {
+    super();
+    this.runPromise = new Promise<Run>((extract) => {
+      this.runPromiseResolver = extract;
+    });
+  }
+
+  async persistRun(run: Run) {
+    this.runPromiseResolver(run);
```
Comment (hinthornw) [inline langchain/src/smith/runner_utils.ts:78]: does this always work for nested runs? Or do we need to check that it's the root trace?

File: langchain/src/smith/tests/run_on_dataset.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+      {
+        evaluatorType: "labeled_criteria",
+        criteria: "correctness",
+        feedbackKey: "correctness",
+        formatEvaluatorInputs: ({
+          rawInput,
+          rawPrediction,
+          rawReferenceOutput,
+        }: any) => {
+          console.log(rawInput);
```
Comment (hinthornw) [inline langchain/src/smith/tests/run_on_dataset.int.test.ts:227]: Still want to keep these?

**Review by hinthornw — APPROVED** [review]

File: langchain/src/smith/tests/run_on_dataset.int.test.ts (hunk 2 above)
Comment (jacoblee93) [reply langchain/src/smith/tests/run_on_dataset.int.test.ts:227]: Sure could remove

File: langchain/src/smith/runner_utils.ts (hunk 1 above)
Comment (jacoblee93) [reply langchain/src/smith/runner_utils.ts:78]: Not sure

**Comment by Gr33nLight** [comment]
Commenting on this so that I know when this is merged.

**Comment by Gr33nLight** [comment]
Noice, was gonna create an issue about this but it seems this is the fix 👍🏻 hope this is released soon.

**Comment by jacoblee93** [comment]
Thanks @Gr33nLight! Am about to ship another version now, will post here when that's live.

**Comment by jacoblee93** [comment]
This is now live in langchain@0.1.16 @Gr33nLight

**Comment by Gr33nLight** [comment]
@jacoblee93 Great! Tested it and it works.

**Pull request merged**